docs/data/date-pickers/custom-components/custom-components.md

```diff
@@ -0,0 +1,79 @@
+---
+product: date-pickers
+title: Date and Time pickers - Custom components
+---
+
+# Custom components
+
+<p class="description">The date picker let the user select a date.</p>
+
+## Overriding components
+
+The picker components allows you to override internal components with the `components` prop.
+
+If you wish to pass additional props in a component slot, you can do it using the `componentsProps` prop.
+
+As an example, you could override the action bar and pass additional props as below.
+
+```jsx
+<DatePicker
+  {...otherProps}
+  components={{
+    ActionBar: CustomActionBar,
+  }}
+  componentsProps={{
+    actionBar: ['clear'],
+  }}
+/>
+```
+
+:::info
+The casing is different between the `components` (ActionBar) and `componentsProps` (actionBar) props.
```

Comment:
To be more specific.

```suggestion
The `components` prop uses Pascal case (`ActionBar`), while `componentsProps` uses camel case (`actionBar`).
```

Reply:
We should probably do the same change on the data grid equivalent page when improving it then